.github/workflows/deployment.yml

```diff
@@ -0,0 +1,79 @@
+name: safe-gelato-relay-service
+
+on: [ push, pull_request ]
+
+jobs:
+
+  docker-publish-staging:
+    if: (github.event_name == 'push' && github.ref == 'refs/heads/main')
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          # Number of commits to fetch. 0 indicates all history for all branches and tags.
+          fetch-depth: '0'
+      - name: Set Version
+        run: |
+          description="$(git describe --tags --always)"
+          echo "VERSION=${description:1}" >> $GITHUB_ENV
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2.4.1
+      - name: Login to DockerHub
+        uses: docker/login-action@v2.1.0
+        with:
+          username: ${{ secrets.DOCKER_USER }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      - name: Build and push
+        uses: docker/build-push-action@v4.0.0
+        with:
+          context: .
+          push: true
+          build-args: VERSION=${{ env.VERSION }}
+          tags: safeglobal/safe-client-gateway:staging
+          cache-from: type=registry,ref=safeglobal/safe-client-gateway:staging
+          cache-to: type=inline
+
+  deploy:
+    runs-on: ubuntu-20.04
+    env:
+      DOCKERHUB_ORG: safeglobal
+      DOCKERHUB_PROJECT: safe-gelato-relay-service
+    needs: [ rustfmt, tests ]
+    if: github.ref == 'refs/heads/develop' || startsWith(github.ref, 'refs/tags/v')
+    steps:
+      - name: Checkout branch
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # checks out all branches and tags, necessary for versioning /about endpoint
+
+      - name: Dockerhub login
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKER_USER }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+      - name: Deploy main
+        if: github.ref == 'refs/heads/main'
+        run: bash scripts/deploy_docker.sh staging
+
+      - name: Deploy tag
+        if: startsWith(github.ref, 'refs/tags/')
+        run: bash scripts/deploy_docker.sh ${GITHUB_REF##*/}
+
+      - name: Deploy develop # unused
+        if: github.ref == 'refs/heads/develop'
+        run: bash scripts/deploy_docker.sh develop
+
+  autodeploy:
+    runs-on: ubuntu-20.04
+    needs: [docker-publish-staging]
+    if: github.event_name == 'push' && github.ref == 'refs/heads/main'
+    steps:
+    - uses: actions/checkout@v3
+    - name: Deploy staging
+      run: bash scripts/autodeploy.sh
```

Comment:
We don't have this file. This is matches the CI pipeline file from the CGW. Why are we reusing all the steps from the CGW (Rust)?